### PR TITLE
Add support to configure HTTP proxy settings in Jenkins Pod

### PR DESCRIPTION
**Pull request by PhilipGough**
ping @laurafitzgerald

### CODE REVIEW

File: configure-buildfarm/tasks/main.yml [hunk 1]
```diff
@@ -6,22 +6,29 @@
...
   name: "Get jenkins url"
   shell: "oc get route jenkins --namespace={{ project_name }} -o jsonpath='{.spec.host}'"
   register: route_output
 
 - pause:
     prompt: |
          The BuildFarm was successfully installed, Do you want to configure your existing RHMAP installation to use the new BuildFarm ?
-         
+
          If you accept to continue please add your current ~/.ssh/id_rsa.pub key here {{ route_output.stdout }}/user/{{ jenkins_username | default("admin") }}/configure
```
Comment (laurafitzgerald) [inline configure-buildfarm/tasks/main.yml]: Could we change this to `If you want to continue,`

Comment (PhilipGough) [inline configure-buildfarm/tasks/main.yml]: We can. It wasn't written in this pr but I'll make the change

File: configure-buildfarm/tasks/main.yml [hunk 2]
```diff
@@ -30,6 +37,26 @@
...
+  register: proxy_task
+  when:
+    - proxy_host is defined
+    - proxy_host != ''
+    - proxy_port is defined
+    - proxy_port != ''
+
+-
+  name: "Configure HTTP Proxy Settings"
+  command: java -jar /tmp/jenkins-cli.jar -s https://{{ route_output.stdout }}  groovy /tmp/proxy-config.groovy
```
Comment (laurafitzgerald) [inline configure-buildfarm/tasks/main.yml]: We can use {{ jenkins_route_protocol }} here. This is now set in the `vars/buildfarm.yml` in productised. Could we add it in the community vars file also?

Comment (PhilipGough) [inline configure-buildfarm/tasks/main.yml]: Done

**Comment by PhilipGough** [comment]
@laurafitzgerald changed based on review, ready for verification when you, or maybe @matskiv  get a chance :)

File: configure-buildfarm/tasks/main.yml [hunk 3]
```diff
@@ -6,30 +6,57 @@
...
+-
   name: "Get jenkins url"
   shell: "oc get route jenkins --namespace={{ project_name }} -o jsonpath='{.spec.host}'"
   register: route_output
 
 - pause:
     prompt: |
          The BuildFarm was successfully installed, Do you want to configure your existing RHMAP installation to use the new BuildFarm ?
-         
-         If you accept to continue please add your current ~/.ssh/id_rsa.pub key here {{ jenkins_route_protocol | default("https")}}://{{ route_output.stdout }}/user/{{ jenkins_username | default("admin") }}/configure
```
Comment (laurafitzgerald) [inline configure-buildfarm/tasks/main.yml]: How come the protocol is being removed here?

**Comment by laurafitzgerald** [comment]
**Verification**
- Added proxy vars to inventory file
- Ran:
```
ansible-playbook -i inventories/engineering/vagrant/vagrant-host playbooks/buildfarm.yml
```
**Expected**
Proxy configuration to be added to Jekins

**Result**
-> Jenkins-> Managed Jenkins -> Manage Plugins -> Advanced
Proxy Configuration Set

Changes Verified.

**Pull request merged**